### PR TITLE
#194 Program Management Follow-up

### PR DESCRIPTION
**Pull request by gitcow**
- Show finalized proposals on selection index.
- Confirmation notes viewable from selection index & show pages.
- Promote & Decline buttons removed from program proposals show page.
- Removed ::text|confirmation_link:: in favor of markdown's [text](::confirmation_link::) style.
- Language tweaks on speaker confirmation flow.
- Block non-program-team staff from the program pages. Refactored speakers, proposals,
  and program_sessions controllers to inherit from program controller.
- Cleaned up role-checking controller code a bit.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -33,7 +30,7 @@ def after_sign_in_path_for(user)
       request.referrer
     elsif session[:target]
       session.delete(:target)
-    elsif event_staff?(current_event)
+    elsif current_user.staff_for?(current_event)
```
Comment (mghaught) [inline app/controllers/application_controller.rb]: This should use the user object passed in

File: app/controllers/staff/teammates_controller.rb [hunk 2]
```diff
@@ -62,4 +62,11 @@ def group_count(group)
     team_counts
   end
 
+  def require_contact_email
+    if @event.contact_email.empty?
```
Comment (mghaught) [inline app/controllers/staff/teammates_controller.rb]: Curious about the use of @event when you've got current_event referenced in the rest of the controller.

**Review by mghaught — CHANGES_REQUESTED** [review]
Looks good.  A minor tweak or two via comments.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (gitcow) [reply app/controllers/application_controller.rb]: Ah, thanks

File: app/controllers/staff/teammates_controller.rb (hunk 2 above)
Comment (gitcow) [reply app/controllers/staff/teammates_controller.rb]: Oh, yep. Looks like I missed that when I copied it over.

**Review by mghaught — APPROVED** [review]

**Pull request merged**